### PR TITLE
feat: add settings for go sql.DB settings

### LINKED ISSUE #11498: outputs.mysql loses connection to sql server after 24-36 hours
### Relevant telegraf.conf

```toml
[[outputs.sql]]
  driver = "mysql"

  data_source_name = "${DB_USER}:${DB_PASSWORD}@tcp(${DB_ADDRESS}:3306)/jenkins?multiStatements=true"
  table_template = "CREATE TABLE {TABLE}({COLUMNS})"
  table_exists_template = "SELECT 1 FROM {TABLE} LIMIT 1"
  init_sql = "SET sql_mode='ANSI_QUOTES';"
```

### Logs from Telegraf

```text
## Application Startup
2022-07-11T17:10:49Z I! Starting Telegraf 1.20.4
2022-07-11T17:10:49Z I! Loaded inputs: jenkins
2022-07-11T17:10:49Z I! Loaded aggregators:
2022-07-11T17:10:49Z I! Loaded processors:
2022-07-11T17:10:49Z I! Loaded outputs: sql
2022-07-11T17:10:49Z I! Tags enabled: host=ip-10-178-133-189.ec2.internal
2022-07-11T17:10:49Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"ip-10-178-133-189.ec2.internal", Flush Interval:10s
2022-07-11T17:10:49Z D! [agent] Initializing plugins
2022-07-11T17:10:49Z D! [agent] Connecting outputs
2022-07-11T17:10:49Z D! [agent] Attempting connection to [outputs.sql]
2022-07-11T17:10:49Z D! [agent] Successfully connected to outputs.sql
2022-07-11T17:10:49Z D! [agent] Starting service inputs
2022-07-11T17:10:59Z D! [outputs.sql] Wrote batch of 27 metrics in 339.444958ms

... some time later ....

[mysql] 2022/07/12 19:07:44 packets.go:123: closing bad idle connection: EOF
[mysql] 2022/07/12 19:07:44 connection.go:173: driver: bad connection
2022-07-12T19:07:44Z E! [outputs.sql] Error during insert: Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '"jenkins_job"("timestamp","result","source","host","name","parents","port","dura' at line 1, INSERT INTO "jenkins_job"("timestamp","result","source","host","name","parents","port","duration","result_code","number") VALUES(?,?,?,?,?,?,?,?,?,?)
```

### System info

telegraf:1.20.4-alpine docker image

### Docker

_No response_

### Steps to reproduce

1. start telegraf with one input and an outputs.sql type of mysql
2. let run for 24 hours or more and observe successfully inserted data in mysql

### Expected behavior

telegraf runs indefinitely (at least consistently more than 24 hours) and successfully writes output to mysql server

### Actual behavior

telegraf logs `closing bad idle connection: EOF` error message, all ensuing `INSERT` statements fail until the telegraf application is restarted

### Additional info

The go-mysql-driver clearly mentions that the ConnMaxLifetime should be set to less than 5 mins to address potential issues like we are seeing - https://github.com/go-sql-driver/mysql#important-settings.  However, it does not appear that this value can be set on the DSN as part of the connection string, and [can only be set by calling `db. SetConnMaxLifetime()`](https://github.com/go-sql-driver/mysql#usage) which we don't appear able to do without modifying the telegraf code.

Note: In our setup, mysql is running as an AWS RDS Aurora Serverless database that is presenting a mysql 5.6.10-compatible interface.

### PR DESCRIPTION
**Pull request by reimda**
resolves #11498

The golang SQL API exposes a few settings related to pooled connections. (see https://pkg.go.dev/database/sql#DB) These settings need to be changed for some SQL drivers to connect reliably to some databases.

This PR adds telegraf settings that mirror the golang sql settings. This allows users to set them through telegraf.conf and be able to connect to databases that require specific settings.

The golang sql settings have defaults and since telegraf didn't set the settings, it was using the defaults. I mirrored the default values in the new telegraf settings. This should make the sql output behave the same as before.

### CODE REVIEW

**Review by MyaLongmire — APPROVED** [review]

**Review by srebhan — COMMENTED** [review]
Thanks @reimda for the fix. I only have a comment about option naming to make it consistent with the `sql` input plugin...

File: plugins/outputs/sql/sample.conf [hunk 1]
```diff
@@ -51,3 +51,17 @@
...
+
+  ## Maximum amount of time a connection may be reused. "0s" means connections
+  ## are never closed due to age.
+  # conn_max_lifetime = "0s"
+
+  ## Maximum number of connections in the idle connection pool. 0 means unlimited.
+  # max_idle_conns = 2
+
+  ## Maximum number of open connections to the database. 0 means unlimited.
+  # max_open_conns = 0
```
[the remark this answers is not in the document]
Comment (reimda) [reply plugins/outputs/sql/sample.conf]: It makes sense to me to make them match. Done in be42fc4

**Review by powersj — APPROVED** [review]
Thanks for jumping on this one!

**Review by srebhan — APPROVED** [review]
Thanks @reimda! Looks good to me.

**Pull request merged**